### PR TITLE
Merge in @Awernx's change to store configuration in XDG directories when ~/.mcfly does not exist

### PR DESCRIPTION
**Pull request by cantino**
Replaces https://github.com/cantino/mcfly/pull/124 and includes some cargo updates and doc changes.

@Awernx, does this all look right?

### CODE REVIEW

**Review by Awernx — APPROVED** [review]

**Comment by Awernx** [comment]
Looks good!

File: src/settings.rs [hunk 1]
```diff
@@ -496,17 +496,38 @@ impl Settings {
...
+        } else {
+            base_dir
+        }
+    }
+
+    fn home_mcfly_exists() -> (bool, PathBuf) {
+        let user_dirs = UserDirs::new().unwrap();
+        let user_dirs_file = user_dirs.home_dir().join(PathBuf::from(".mcfly"));
+
+        (user_dirs_file.exists(), user_dirs_file)
```
Comment (praveenperera) [inline src/settings.rs:531]: Maybe returning an Option would be better here.

File: src/settings.rs [hunk 2]
```diff
@@ -496,17 +496,38 @@ impl Settings {
...
+    }
+
+    fn mcfly_base_path(base_dir: PathBuf) -> PathBuf {
+        let (home_exists, mcfly_path) = Settings::home_mcfly_exists();
+
+        if home_exists {
+            mcfly_path
+        } else {
+            base_dir
+        }
```
Comment (praveenperera) [inline src/settings.rs:524]: If we return an option for `home_mcfly_exists` we can just us an `unwrap_or` here.

**Review by praveenperera — COMMENTED** [review]
Small nits, looks good.

**Comment by cantino** [comment]
How's this @praveenperera?

**Comment by praveenperera** [comment]
Looks good

**Comment by cantino** [comment]
Thanks everyone

**Comment by cantino** [comment]
https://github.com/cantino/mcfly/releases/tag/v0.6.0 released!

**Pull request merged**